packages/ui/src/components/Visualization/Canvas/expression.service.ts

```diff
@@ -0,0 +1,146 @@
+  }
+
+  /**
+   * Get the language schema from the language catalog. ATM it delegates to {@link CamelComponentSchemaService.getSchemaFromCamelCommonProperties}.
+   * @TODO The language `xtokenize` has `namespace` property with the type `array`, which causes an error in uniforms. Fix the schema here, or in {@link CamelComponentSchemaService.getSchemaFromCamelCommonProperties} if it could be common.
+   * @param languageCatalog
+   */
+  static getLanguageSchema(languageCatalog: ICamelLanguageDefinition) {
+    return CamelComponentSchemaService.getSchemaFromCamelCommonProperties(languageCatalog.properties);
+  }
+
+  /**
+   * Parse the expression model from the parent step model object. Since expression has several dialects,
+   * this method tries to read all possibility and merge into a single representation.
+   * @param languageCatalogMap
+   * @param parentModel
+   */
+  static parseExpressionModel(
+    languageCatalogMap: Record<string, ICamelLanguageDefinition>,
+    parentModel: Record<string, unknown>,
+  ): {
+    language: ICamelLanguageDefinition | undefined;
+    model: Record<string, unknown>;
+  } {
+    let languageModelName = 'simple';
+    let model = undefined;
+    if (parentModel?.expression && Object.keys(parentModel.expression).length > 0) {
+      languageModelName = Object.keys(parentModel.expression)[0];
+      model = ExpressionService.parseLanguageModel(
+        parentModel.expression as Record<string, unknown>,
+        languageModelName,
+      );
+    } else {
+      for (const language of Object.values(languageCatalogMap)) {
+        if (parentModel[language.language.modelName]) {
+          languageModelName = language.language.modelName;
+          model = ExpressionService.parseLanguageModel(parentModel, language.language.modelName);
+          break;
+        }
+      }
+      if (!model) {
+        parentModel.expression = {};
+        model = {};
+        (parentModel.expression as Record<string, unknown>)[languageModelName] = model;
+      }
+    }
+    const language = this.getDefinitionFromModelName(languageCatalogMap, languageModelName);
+    return { language, model };
+  }
+
+  private static parseLanguageModel(model: Record<string, unknown>, langName: string) {
+    const lang = model[langName];
+    if (typeof lang === 'string') {
+      return { expression: lang };
+    } else {
+      return lang as Record<string, unknown>;
+    }
+  }
+
+  private static getDefinitionFromModelName(
+    languageCatalogMap: Record<string, ICamelLanguageDefinition>,
+    modelName: string,
+  ): ICamelLanguageDefinition | undefined {
+    return Object.values(languageCatalogMap).find((model) => model.language.modelName === modelName);
+  }
+
+  /**
+   * Set the expression model to the parent step model object. This method uses the most verbose dialect, i.e.
+   * ```yaml
+   * - setBody:
+   *     expression:
+   *       simple:
+   *         expression: ${body}
+   *         trim: true
+   * ```
```

Comment:
This is great, thanks for providing the example

Reply:
mmm, could be possible to point also in the example, what the params represent based on the YAML example?

Reply:
Do you mean we want an explanation for e.g. `trim` parameter in this comment?

Reply:
Ah, no, I meant the function parameters.

```
@param parentModel
@param languageModelName
@param newExpressionModel
```

For instance, are those parameters related to the YAML example? do they correspond to a particular field? if that's the case, can we point out which ones?

Reply:
ahh OK, that makes sense, I'll follow up